### PR TITLE
Fix Light mode screen issues.

### PR DESCRIPTION
**Pull request by OrionsGuardian**
### Prerequisites

- [ ] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

Extended from some earlier work that was done to create a light mode for Constellation.
A number of screens have been further updated for light mode usage.

### Why Should This Be In Core?

Core UI enhancement when using a light mode Look and Feel.

### Benefits

The UI will have a more appropriate and consistent colour scheme when a light mode Look and Feel is being used.

### Verification Process

Change to a Light mode Look and Feel, such as "FlatLaf Light" and confirm that the screen elements are appropriately coloured for light mode usage.
Confirm the following changes:
Attribute Editor: Default lighter colours for each attribute element, and light background area for each element value.
Attribute Editor: Icon Editor dialog, background now allows white icons to be visible.
Data Access View: Workflow Options: Black version of icons now used in the menu.
Conversation View: Light grey background to allow white icons to be visible, and conversation bubbles have slightly brighter colour.
Notes View: Black text at the top of the note, and lighter default background colour.
Error Report View: Adjusted colour and sizing of components.

### Applicable Issues

#2016

### CODE REVIEW

File: CoreErrorReportView/src/au/gov/asd/tac/constellation/views/errorreport/ErrorReportTopComponent.java [hunk 1]
```diff
@@ -774,79 +791,149 @@ private List<String> getErrorLevelList(final boolean unAcknowledgedEntriesOnly)
...
+                redIncrement = -4;
+                greenBase = 220;
+                greenIncrement = -11;
+                blueBase = 220;
+                blueIncrement = -11;
+            }
+                        
         }
-        
+        String areaBackgroundColour = "radial-gradient(radius 100%, " + backgroundColour + " 0%, " + backgroundFadeColour + " 100%)";
```
Comment (Delphinus8821) [inline CoreErrorReportView/src/au/gov/asd/tac/constellation/views/errorreport/ErrorReportTopComponent.java:925]: final

**Review by Delphinus8821 — CHANGES_REQUESTED** [review]
Looks good apart from this weird mouse hover behaviour happening on the Icon Editor dialog in the Attribute Editor which only occurs in Light Mode.
image

**Review by Delphinus8821 — APPROVED** [review]

File: CoreErrorReportView/src/au/gov/asd/tac/constellation/views/errorreport/ErrorReportDialog.java [hunk 2]
```diff
@@ -167,6 +179,15 @@ public ErrorReportDialog(final ErrorReportEntry errorEntry) {
         blockRepeatsCheckbox.setSelected(errorEntry.isBlockRepeatedPopups());
         buttonPane.setCenter(blockRepeatsCheckbox);
         buttonPane.setRight(closeButton);
+        
+        root.widthProperty().addListener(new ChangeListener<Number>(){
+            @Override
+            public void changed(ObservableValue<? extends Number> observable, Number oldValue, Number newValue) {
```
Comment (antares1470) [inline CoreErrorReportView/src/au/gov/asd/tac/constellation/views/errorreport/ErrorReportDialog.java:185]: can this be made a lambda (if not, parameters can be final)

File: CoreErrorReportView/src/au/gov/asd/tac/constellation/views/errorreport/ErrorReportEntry.java [hunk 3]
```diff
@@ -64,13 +64,20 @@ public String getTrimmedHeading(final int length) {
         if (heading == null) {
             return "-NULL-";
         }
-        if (heading.length() < length) {
-            return heading;
+
+        String adjustedHeading = heading;
+        if (adjustedHeading.endsWith("\n")) {
```
Comment (antares1470) [inline CoreErrorReportView/src/au/gov/asd/tac/constellation/views/errorreport/ErrorReportEntry.java:69]: `SeparatorConstants.NEWLINE`

**Review by antares1470 — CHANGES_REQUESTED** [review]
A couple of things from testing:
1) For the Attribute editor, I think the blue for the custom attributes blends isn't distinct enough from the grey. Perhaps a darker blue might help? I would be saying similar things about the grey for the hidden attributes but I think the white text offers enough contrast.
attributeEditorLightColours

2) Probably doesn't need to be addressed here but figured it worth a mention. With changing the default color for notes in the Notes View, its worth noting that when saved, they are saved with that color and so if you switch to dark mode later, you still have the light colored note (and potentially a series of lighter and darker user notes if you don't change the colors manually)

**Review by antares1470 — APPROVED** [review]
Having played around with it a little, I think there is possibly some further areas to improve the light/dark mode compatibility. One of the main issues is that a number of visual parts of the graph are saved with the graph and so this makes switching between light and dark mode interesting when your graphs don't really do the same thing (opening an old graph in light mode still has the dark background).

That being said, that's a much bigger potential body of work, and not one that needs to be addressed here

**Pull request merged**